QA v1.7.0-rc22

EZheln opened the pull request:
Updated tests according to requirements
- change selector for logs content on Log tab - Jobs and Workflows
- change the order of Overview_Headers - Jobs and Workflows
- change the text on popup Delete_Scheduled_Job - Jobs and Workflows 
- changed selectors for Table_Refresh_Button, Table_Expand_Rows_Button, New_Function_Button, Table_Name_Filter_Input, deleted Show_Untagged_Functions_Checkbox - ML Function ML-5846
- changed overview headers - Models
- added 'Delete all' in details action menu - Models
- changed title 'Model cannot be deployed' pop-up - Models
- changed selector for Labels_Field - Models
- changed artifact path in default project - Project Monitoring
- changed selectors and warning text for Secrets tab - Project Settings
- changed warning text to 'A project named "[^"]+[$"] already exists' - Projects
- changed text for Archive project - Projects
- changed text for delete project - Projects
- changed text for limit project - Projects
- updated summary structure
- updated overwrite message - Artifacts
- changed path from redisnosql to nosql - Feature Store
- added Metrics tab - Models
- ML Function - updated the tests with steps to achieve the testable element
- Project Settings - changed selector for Secret input key/value
- browser configs - block the pop-up windows in chrome to achieve the testable element
- synthetic data - added data for scheduled pipeline
- mock - added logic for updating values in files with synthetic data
- mock - created getProjectsSummaries filteting by statuses and time for jobs, pipelines, sheduled jobs and pipelines
- mock - created get runs for cross projects Monitoring page with filteting by statuses and time

Commit:
Updated tests according to requirements:
- change selector for logs content on Log tab - Jobs and Workflows
- change the order of Overview_Headers - Jobs and Workflows
- change the text on popup Delete_Scheduled_Job - Jobs and Workflows
- changed selectors for Table_Refresh_Button, Table_Expand_Rows_Button, New_Function_Button, Table_Name_Filter_Input, deleted Show_Untagged_Functions_Checkbox - ML Function ML-5846
- chenged overview headers - Models
- added 'Delete all' in details action menu - Models
- changed title 'Model cannot be deployed' pop-up - Models
- changed selector for Labels_Field - Models
- changed artifact path in default project - Project Monitoring
- changed selectors and warning text for Secrets tab - Project Settings
- changed warning text to 'A project named "[^"]+[$"] already exists' - Projects
- changed text for Archive project - Projects
- changed text for delete project - Projects
- changed text for limit project - Projects
- updated summary structure
- updated overwrite message - Artifacts
- changed path from redisnosql to nosql - Feature Store
- added Metrics tab - Models

Taras-Hlukhovetskyi commented on tests/mockServer/dateSynchronization.js:
```diff
@@ -0,0 +1,105 @@
+        }
+
+        // update start_time та last_update
+        const updatedRuns = runsData.runs.map(run => {
```
You can use such condition instead of many if-else (except the last one with 'oneHourAgo'), I am not sure if we need to check name if we check the uid
`if (['cf842616c89347c7bb7bca2c9e840a21', '76f48c8165da473bb4356ef7b196343f' ,'...' ].includes(run.metadata.uid)) {
  return {
                    ...run,
                    status: {
                        ...run.status,
                        start_time: formatDate(oneDayAgo),
                        last_update: formatDate(oneDayAgo)
                    }
                }
}`

Taras-Hlukhovetskyi commented on tests/mockServer/mock.js:
```diff
@@ -514,6 +518,80 @@ function deleteSecretKeys(req, res) {
+
+  // Runs
+  const projectData = runs.runs
+    .filter(run => run.kind === 'run' && new Date(run.status.last_update).getTime() > last24Hours.getTime())
```
you can remove .getTime() as for piplines, also here and for pipelines need to check if a time not bigger that 'now'

Taras-Hlukhovetskyi commented on tests/mockServer/dateSynchronization.js:
```diff
@@ -0,0 +1,105 @@
+    const oneHourAgo = new Date(now.getTime() - (45 * 60 * 1000))
+
+    // Past 24h
+    const oneDayAgo = new Date(now.getTime() - (20 * 60 * 60 * 1000))
```
20 hours before

Taras-Hlukhovetskyi commented on tests/mockServer/dateSynchronization.js:
```diff
@@ -0,0 +1,105 @@
+    const now = new Date()
+
+    // Past hour
+    const oneHourAgo = new Date(now.getTime() - (45 * 60 * 1000))
```
45 minutes before

Taras-Hlukhovetskyi commented on tests/mockServer/dateSynchronization.js:
```diff
@@ -0,0 +1,105 @@
+const fs = require('fs')
```
use imports

Taras-Hlukhovetskyi commented on tests/mockServer/dateSynchronization.js:
```diff
@@ -0,0 +1,105 @@
+    })
+}
+
+module.exports = updateRuns
```
remove if use import and empty line at the end of file

Commit:
Updated tests according to requirements:
- ML Function - updated the tests with steps to achieve the testable element
- Project Settings - changed selector for Secret input key/value
- browser configs - block the pop-up windows in chrome to achieve the testable element
- synthetic data - added data for scheduled pipeline
- mock - added logic for updating values in files with synthetic data
- mock - created getProjectsSummaries filtering by statuses and	time for jobs, pipelines, scheduled jobs and pipelines
- mock - created get runs for cross projects Monitoring page  with filtering by statuses and time
